docs/software-platforms/docker/docker-images.md

```diff
@@ -0,0 +1,114 @@
+
+You will also need to have [Docker installed on your computer](https://docs.docker.com/engine/install/) to build and test the images.
+
+## Docker template files
+
+When you create an analysis module, a template `Dockerfile` and a `.dockerignore` file will be created in the root of the analysis module.
+
+The  `Dockerfile` includes a `FROM` statement that specifies the base image to use and a couple of `LABEL` statements to populate metadata about the image.
+
+The `.dockerignore` file is used to specify files and directories that should *not* be included in the Docker build environment.
+By default, this will be set to ignore all files with the exception of environment files (e.g., `renv.lock`, `environment.yml`, `conda-lock.yml`).
+This will keep the build environment small and prevent unnecessary files from being included in the image.
+
+## Base images
+
+The base image you choose will depend on the software requirements of your analysis module.
+You should aim to use a publicly available base image that is in active development and well-maintained in order to ensure that your image is secure and up-to-date.
+You should always use a *versioned* base image to ensure that your analysis module is reproducible, i.e., not the one tagged as `latest`.
+
+## Additional software
+
+If your analysis module requires additional software that is not available in the base image, you will need to install it in the Dockerfile.
+The easiest way to do this is likely to be to use the same tool (`renv`, `conda`, etc.) that you used to define the software environment for your analysis module.
+If necessary, you can also include statements to install other software using the system package manager (e.g., `apt-get` for Debian-based images).
+
+## R-based images
+
+If you are working with R, you will likely want to use one of the [`rocker/r-ver` images](https://hub.docker.com/r/rocker/r-ver/tags) or a[`bioconductor/r-ver` image](https://hub.docker.com/r/bioconductor/r-ver/tags) images, which are a set of Docker images that provide complete R environments, tied to a specific version of R and/or Bioconductor.
+The Bioconductor images are especially handy as they include system software dependencies for Bioconductor packages that may not be installed in the `rocker` images, and support binary installation of Bioconductor packages, which can dramatically speed up build times.
+
+You will want to include the following steps in your Dockerfile:
+
+- Install `renv` (and `remotes` if you have any GitHub-based packages in your environment).
+- Copy the `renv.lock` file from the host environment to the image.
+- Use `renv::restore()` to reproduce the R environment from the `renv.lock` file.
+
+
+A simple `Dockerfile` for an R-based analysis module that uses `renv` for its environment might look like this:
```

Comment:
what about `LABEL`? if it's in the template, it should be here too

Reply:
I wanted to focus on the mechanics with these examples. The labels are long and distracting.